nimpy.nim

```diff
@@ -899,6 +924,32 @@ proc newPyObject(o: PPyObject): PyObject =
     incRef o
     newPyObjectConsumingRef(o)
 
+proc checkPyObjPyType(o: PPyObject): PyBaseType =
+    ## returns the correct PyTypeObject of the given PyObject extracted
+    ## by manually checking all types
+    ## If no call to `returnIfSubclass` returns from this proc, the
+    ## default value of `pbUnsupported` will be returned
+    template returnIfSubclass(pyt, nimt: untyped): untyped =
+        if checkObjSubclass(o, pyt):
+            return nimt
+
+    # check int types first for backward compatibility with Python2
+    returnIfSubclass(Py_TPFLAGS_INT_SUBCLASS or Py_TPFLAGS_LONG_SUBCLASS, pbLong)
+
+    let checkTypes = { pyLib.PyFloat_Type : pbFloat,
+                       pyLib.PyComplex_Type : pbComplex,
+                       pyLib.PyBytes_Type : pbString,
+                       pyLib.PyUnicode_Type : pbString,
+                       pyLib.PyList_Type : pbList,
+                       pyLib.PyTuple_Type : pbTuple,
+                       pyLib.PyDict_Type : pbDict }.toTable
```

Comment:
No need in `toTable` here, and `checkTypes` can be const.

Reply:
Well, I guess I can iterate over the tuples and get the fields, true. It can't be const though, because `pyLib` isn't defined at compile time.